### DISCUSSION TITLE
Best Deployment Method for Milvus on a Single Server

### DISCUSSION THREAD

**Discussion by kietheros** [milvus-io/milvus#37040] [Q&A and General discussion] [answered]
I want to deploy Milvus on a single server. My system already has Kafka, Minio, and etcd installed.
Could you recommend the most stable deployment method?

- Docker Compose for Milvus Standalone:
  - Only deploy Milvus services, configuring it to use external Kafka, Minio, and etcd.
  - Deploy all-in-one (including internal Minio, etcd, and Pulsar on some RAID-1 disks).
- One-Node Cluster (K3s) using Helm:
  - Only deploy the Milvus service, using external Kafka, Minio, and etcd.
  - Deploy all components.

**Comment by yhmo** [comment, marked as answer]
Helm is recommended, cluster mode is recommended. If you use helm to deploy a milvus standalone, it is no much difference with docker-compose standalone.

**Reply by kietheros** [reply]
Thank you for your reply. I can only deploy Milvus on a single server, using Helm is still easier to maintain.

**Reply by xiaofan-luan** [reply]
For single server, docker compose is good enough I guess.
Helm of course works as well.
